### PR TITLE
fix(read_buffer): Avoid panic when creating stats for entirely null columns

### PR DESCRIPTION
**Pull request by alamb**
Resolves https://github.com/influxdata/influxdb_iox/issues/2158
Resolves https://github.com/influxdata/conductor/issues/408

# Rationale

Previously, if a column of all NULL was inserted into the read buffer and then `Chunk::table_summary()` was called, the code would panic:

```
---- chunk::test::table_summaries stdout ----
thread 'chunk::test::table_summaries' panicked at 'unsupported type of stats in read buffer: (Null, Null)', read_buffer/src/table.rs:744:26
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

# Changes

1. Implement statistics conversion for NULL values
2. Remove redundant encoding of Null by removing `OwnedValue::Null` and using `OwnedValue::Scalar(Scalar::Null)` instead
3. Removed `match .. _` catch all match so when new types are added, compiler will ensure new types are handled.

### CODE REVIEW

File: read_buffer/src/table.rs [hunk 1]
```diff
@@ -705,6 +705,9 @@ impl MetaData {
                             distinct_count,
                         })
                     }
+                    (OwnedValue::String(_), mismatch) => {
```
Comment (alamb) [inline read_buffer/src/table.rs:708]: I added these explicit matches so as to remove `_` catchall match which I think meant the compiler didn't catch not handling the missing `NULL` case

File: read_buffer/src/table.rs [hunk 2]
```diff
@@ -722,29 +728,57 @@ impl MetaData {
...
+                                column_meta,
+                            );
+                            assert_eq!(
+                                distinct_count,
+                                std::num::NonZeroU64::new(1),
+                                "distinct count for all null was not 1: {:?}",
+                                column_meta,
+                            );
+
+                            make_null_stats(total_count, &column_meta.logical_data_type)
```
Comment (alamb) [inline read_buffer/src/table.rs:767]: here is the actual fix

File: read_buffer/src/value.rs [hunk 3]
```diff
@@ -1138,23 +1138,24 @@ impl std::fmt::Display for &Scalar {
 
 #[derive(Debug, PartialEq, PartialOrd, Clone)]
 pub enum OwnedValue {
-    // Represents a NULL value in a column row.
-    Null,
```
Comment (alamb) [inline read_buffer/src/value.rs:1142]: this was redundant with `Scalar(Scalar::Null)` so I unified the two representations

**Review by NGA-TRAN — APPROVED** [review]
LGTM. I hit this earlier this week asa consequence of a different bug

**Comment by alamb** [comment]
FYI @e-dard  this is a change to the read buffer to unblock getting tools working again -- I don't think it should be controversial but I wanted to ping you in case you wanted to review posthumously. I am happy to update / rework anything if you would like

**Pull request merged**